gflownet/utils/batch.py

```diff
@@ -536,18 +551,26 @@ def _compute_parents(self):
         self.parents_available is set to True.
         """
         self.parents = []
+        self.parents_indices = []
         indices = []
         # Iterate over the trajectories to obtain the parents from the states
         for traj_idx, batch_indices in self.trajectories.items():
             # parent is source
             self.parents.append(self.envs[traj_idx].source)
+            # there's no source state in the batch
+            self.parents_indices.append(-1)
             # parent is not source
             # TODO: check if tensor and sort without iter
             self.parents.extend([self.states[idx] for idx in batch_indices[:-1]])
+            self.parents_indices.extend([idx for idx in batch_indices[:-1]])
```

Comment:
No need to iterate over the indices in this case:
```suggestion
            self.parents_indices.extend(batch_indices[:-1])
```